src/tasks/purchase/taskpurchase.cpp

```diff
@@ -22,9 +22,11 @@ Logger logger("TaskPurchase");
 
 #ifdef MZ_IOS
 // static
-TaskPurchase* TaskPurchase::createForIOS(const QString& receipt) {
+TaskPurchase* TaskPurchase::createForIOS(const QString& receipt,
+                                         bool isOlderOS) {
```

Comment:
The meaning of `older` is going to change over time, I would recommend that we provide a version number here and the implementation of this method can decide which API to use.

Reply:
Doing this as `if (@available(iOS 15, *))` feels more in line with iOS coding norms, rather than pulling the OS version and hoping that format never changes.

I would *never* want something like `isOlderOS` to be a variable used in an API or anything, but in this case it's contained fully within `TaskPurchase` (and it's sole call site, `iosiaphandler.mm`). It changes which API we call within `TaskPurchase`. It also won't be there for long; I'd expect we'll move off iOS 14 within a year or so. If there is eventually a v3 of the API, it will be trivial to change the now-modern-API to be the one that is used when `isOlderOS` is true.*

We could have `TaskPurchase` decide which API to use based on parsing a version number that is passed to it. That seems far more brittle than a boolean, though - and I'm inclined to use a boolean unless there is a good reason not to.

How would you feel about a naming update like `doesUseIOS14API`? Is that too specific?

* isOlderOS is only `true` when the user is on iOS 14. iOS 15 and above uses the newer style. Two notes about this -
1. I considered having a discussion about whether we should stop supporting iOS 14 at this point because of this - and change out the code to only support StoreKit2 (rather than have this forking logic). I figured it wasn't too hard to have this forking logic, and so kept it for now.
2. iOS 14 is the oldest version of iOS we support. It came out 4 years ago. I'd imagine that within the next year or so, we'll stop supporting it. (At that point, we can remove a bunch of this code, including the `isOlderOS` variable.)
3. Thus, If we ever have a v3 of this API, we almost certainly will have stopped supporting iOS 14 by then, and we'll again 2 branches for the forking logic.

Reply:
Sorry for the essay. I'm open to talking it out or doing it the way you're considering. It seems to me that it will be brittle, and given the short-term use case of this forking logic across 2 classes, a boolean feels most appropriate and least bug-prone. Thoughgts?